src/braket/devices/local_simulator.py

```diff
@@ -230,21 +240,8 @@ def registered_backends() -> set[str]:
         """
         return set(_simulator_devices.keys())
 
-    def _run_internal_wrap(
-        self,
-        task_specification: Union[
-            Circuit, Problem, OpenQASMProgram, AnalogHamiltonianSimulation, SerializableProgram
-        ],
-        shots: Optional[int] = None,
-        inputs: Optional[dict[str, float]] = None,
-        *args,
-        **kwargs,
-    ) -> Union[GateModelQuantumTaskResult, AnnealingQuantumTaskResult]:  # pragma: no cover
-        """Wraps _run_interal for pickle dump"""
-        return self._run_internal(task_specification, shots, inputs=inputs, *args, **kwargs)
-
     @singledispatchmethod
-    def _get_simulator(self, simulator: Union[str, BraketSimulator]) -> LocalSimulator:
+    def _get_simulator(self, simulator: Any) -> BraketSimulator:
```

Comment:
shouldn't we keep the type hint for `simulator` as `Union[str, BraketSimulator]`?

Reply:
I prefer to keep the specific type hints to the implementations; otherwise, the base `singledispatch` method will have unions that grow with each implementation.

Reply:
In fact, I'd prefer to keep the base dispatch signatures blank if they're variable, but I don't think there's a way to tell the linter to know that

Reply:
That's fair - I can't find any guidance on which way is preferred, so optimizing for lower maintenance cost makes sense.